### PR TITLE
FR-14228 - Added Step up support for Vue 3

### PR DESCRIPTION
**Pull request by doregg**
Vue 3 SDK implementation to step up.
During working on it we realized multiple gaps with working with vue-router. We fixed most of them and open a ticket for the rest.

One of the fixes was related to the behavior after unmounting (using vue router call unmount and we are not subscribed to the store changes like entitlements and step up, etc...). The fix was to cancel the unsubscribing and handle situation when the dev choose to unsubscribe.
  ```
// Using for managing the store resubscription / unsubscribing to make sure we have only one subscription. Subscribe only when store unsubscribed
  if (store.subscribed) {
    // David suggestion: return the previous state and unsubscribe function if the store is already subscribed to fix unsubscribe issue when working with router-view
```

### CODE REVIEW

File: packages/vue/package.json [hunk 1]
```diff
@@ -22,7 +22,7 @@
     "build:unpkg": "cross-env NODE_ENV=production rollup --config rollup.config.js --format iife"
   },
   "dependencies": {
-    "@frontegg/js": "6.165.0",
+    "@frontegg/js": "6.167.0",
```
Comment (doregg) [inline packages/vue/package.json:25]: merge new js before merging

File: packages/vue/src/auth/mapAuthState.ts [hunk 2]
```diff
@@ -106,12 +107,20 @@ export const mapSsoActions = <K extends keyof SSOActions>(action: K): SSOActions
 export const mapTeamActions = <K extends keyof TeamActions>(action: K): TeamActions[K] => actionGetter(action);
 export const mapTenantsActions = <K extends keyof TenantsActions>(action: K): TenantsActions[K] => actionGetter(action);
 
-export const connectFronteggStoreV3 = (store: EnhancedStore) => {
+export const connectFronteggStoreV3 = (store: CachedEnhancedStore) => {
```
Comment (yuvalotem1) [inline packages/vue/src/auth/mapAuthState.ts:110]: we change the type but not the value?

Comment (doregg) [reply packages/vue/src/auth/mapAuthState.ts:110]: Yup, we override it in this function

File: packages/vue/src/auth/mapAuthState.ts [hunk 3]
```diff
@@ -106,12 +107,20 @@ export const mapSsoActions = <K extends keyof SSOActions>(action: K): SSOActions
...
   const initialState = store.getState();
 
+  // Using for managing the store resubscription / unsubscribing to make sure we have only one subscription. Subscribe only when store unsubscribed
+  if (store.subscribed) {
+    // David suggestion: return the previous state and unsubscribe function if the store is already subscribed to fix unsubscribe issue when working with router-view
+    return { authState: store.previousAuthState, unsubscribe: store.previousUnsubscribe };
```
[the remark this answers is not in the document]
Comment (doregg) [reply packages/vue/src/auth/mapAuthState.ts:116]: Good catch. Will fix it and try to simulate by unsubscribing

**Review by frontegg-david — APPROVED** [review]

File: packages/vue/src/auth/mapAuthState.ts [hunk 4]
```diff
@@ -121,7 +130,12 @@ export const connectFronteggStoreV3 = (store: EnhancedStore) => {
     });
   });
 
-  return { authState, unsubscribe };
+  const unsubscribeWrapper = () => {
+    store.subscribed = false;
+    unsubscribe();
+  };
+
```
Comment (frontegg-david) [inline packages/vue/src/auth/mapAuthState.ts:137]:
```suggestion
  store.previousAuthState = authState;
  store.previousUnsubscribe = unsubscribeWrapper;
```

**Review by frontegg-david — APPROVED** [review]

**Review by yuvalotem1 — APPROVED** [review]
I would also test with masha the client implementation with watchEffect

**Pull request merged**